### PR TITLE
[Fix #841] Fixed the complex class issue of NewsListAdapter

### LINKED ISSUE #841: Class NewsListAdapter is overly complex with 92 cyclomatic complexity
There are 20 if-else statements in the method com.github.mobile.ui.user.NewsListAdapter.update(). Got 92 cyclomatic complexity in class NewsListAdapter when analyzing the code using Android Studio Inspection with Java Class metrics.

### PR DESCRIPTION
**Pull request by wubin28**
I like the integration tests of this project. I think it would be better if the gradle configuration of this project could support running unit tests. I inspected the code of this project using Android Studio and found the class NewsListAdapter was overly complex. The cyclomatic complexity of this class is 92. I want to add a unit test for this class and then refactor it. Unit tests are running faster than integration tests without connecting to a real Android phone or an emulator. After referring to the blog (https://sites.google.com/a/android.com/tools/tech-docs/unit-testing-support), I changed the version of Android gradle plugin to be 1.1.0 and added junit in the dependencies. I added 18 unit tests for class IconAndViewTextManager. Now both the unit test and the original integration tests can be run smoothly in the Android Studio.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -44,6 +44,8 @@ repositories {
 dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
     compile 'com.android.support:appcompat-v7:21.0.3'
+    testCompile 'junit:junit:4.12'
+    testCompile "org.mockito:mockito-core:1.9.5"
     compile 'org.roboguice:roboguice:2.0'
```
Comment (fadils) [inline app/build.gradle]: Could you please put the `testCompile` at the bottom? after all `compile` statements?

File: app/src/test/java/com/github/mobile/ui/user/NewsListAdapterTest.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class NewsListAdapterTest {
+
+    @Test
+    public void testUpdate() throws Exception {
+        assertEquals(1, 1);
```
Comment (fadils) [inline app/src/test/java/com/github/mobile/ui/user/NewsListAdapterTest.java]: What exactly are we testing here?

**Comment by wubin28** [comment]
Refactored 17 if-else-if statements into enum constant-specific classes so that the complex class issue of class NewsListAdapter was fixed.

File: app/build.gradle (hunk 1 above)
Comment (wubin28) [inline app/build.gradle]: Done.

File: app/src/test/java/com/github/mobile/ui/user/NewsListAdapterTest.java (hunk 2 above)
Comment (wubin28) [inline app/src/test/java/com/github/mobile/ui/user/NewsListAdapterTest.java]: Removed this test and added other 18 unit tests to test the icon and
formatted text for each event. .

**Comment by larsgrefer** [comment]
fixes #733

File: app/src/main/java/com/github/mobile/ui/user/EventType.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
+package com.github.mobile.ui.user;
+
+import com.github.mobile.ui.StyledText;
+import com.github.mobile.util.TypefaceUtils;
+
+import org.eclipse.egit.github.core.event.Event;
+import org.eclipse.egit.github.core.event.IssuesPayload;
+
+/**
+ * Created by twer on 3/22/15.
```
Comment (hzsweers) [inline app/src/main/java/com/github/mobile/ui/user/EventType.java]: Let's have a descriptive doc or just remove the file header

**Comment by wubin28** [comment]
Removed the empty file header.

**Comment by hzsweers** [comment]
@wubin28 we just merged a fairly significant PR that updates the project structure. If you're willing to update the PR to be mergable with that, I'd be happy to merge it

**Comment by wubin28** [comment]
@hzsweers Just got that. Let me have a try. Thx.

**Comment by wubin28** [comment]
Merged and fixed conflicts.

**Pull request merged**